test/e2e/scale_test.go

```diff
@@ -0,0 +1,151 @@
+				WorkerPerMachineDeploymentCount: ptr.To[int64](1),
+				// MachineDeploymentCount can be overwritten via `CAPI_SCALE_MACHINE_DEPLOYMENT_COUNT`.
+				MachineDeploymentCount: ptr.To[int64](3),
+				// AdditionalClusterClassCount can be overwritten via `CAPI_SCALE_ADDITIONAL_CLUSTER_CLASS_COUNT`.
+				AdditionalClusterClassCount: ptr.To[int64](4),
+				// DeployClusterInSeparateNamespaces can be overwritten via `CAPI_SCALE_DEPLOY_CLUSTER_IN_SEPARATE_NAMESPACES`.
+				DeployClusterInSeparateNamespaces: ptr.To(true),
+				// UseCrossNamespaceClusterClass can be overwritten via `CAPI_SCALE_USE_CROSS_NAMESPACE_CLUSTER_CLASS`.
+				UseCrossNamespaceClusterClass: ptr.To(false),
+
+				// The runtime extension gets deployed to the test-extension-system namespace and is exposed
+				// by the test-extension-webhook-service.
+				// The below values are used when creating the cluster-wide ExtensionConfig to refer
+				// the actual service.
+				ExtensionServiceNamespace: "capv-test-extension",
+				ExtensionServiceName:      "capv-test-extension-webhook-service",
+				ExtensionConfigName:       "capv-scale-test-extension",
+
+				PostNamespaceCreated: func(managementClusterProxy framework.ClusterProxy, workloadClusterNamespace string) {
+					testSpecificSettingsGetter().PostNamespaceCreatedFunc(managementClusterProxy, workloadClusterNamespace)
+
+					// Overwrite some variables with placeholders which get replaced later on in PostScaleClusterNamespaceCreated
+					testSpecificVariables := map[string]string{
+						"VSPHERE_SERVER":              scaleClusterVSphereServerPlaceholder,
+						"VSPHERE_TLS_THUMBPRINT":      scaleClusterVSphereTLSThumbprintPlaceholder,
+						"CONTROL_PLANE_ENDPOINT_IP":   scaleClusterControlPlaneEndpointIPPlaceholder,
+						"CONTROL_PLANE_ENDPOINT_PORT": scaleClusterControlPlaneEndpointPortPlaceholder,
+						// Cluster API uses a hard coded name for the ExtensionConfig.
+						"RUNTIME_SDK_EXTENSIONCONFIG_NAME": "capv-scale-test-extension",
+					}
+					// Re-write the clusterctl config file and add the new variables created above.
+					Byf("Writing a new clusterctl config to %s", testSpecificSettingsGetter().ClusterctlConfigPath)
+					Expect(clusterctl.CopyAndAmendClusterctlConfig(ctx, clusterctl.CopyAndAmendClusterctlConfigInput{
+						ClusterctlConfigPath: testSpecificSettingsGetter().ClusterctlConfigPath,
+						OutputPath:           testSpecificSettingsGetter().ClusterctlConfigPath,
+						Variables:            testSpecificVariables,
+					})).To(Succeed())
+				},
+
+				PostScaleClusterNamespaceCreated: func(clusterProxy framework.ClusterProxy, clusterNamespace string, clusterName string, clusterClassYAML []byte, clusterTemplateYAML []byte) ([]byte, []byte) {
+					// Run additional initialization required for the namespace.
+					if testMode == SupervisorTestMode {
+						switch testTarget {
+						case VCenterTestTarget:
+							setupNamespaceWithVMOperatorDependenciesVCenter(clusterProxy, clusterNamespace)
```

Comment:
This is unreachable code, right? (the whole test only works with vcsim)

(but fine for me to keep it, if you prefer)

Reply:
I'd prefer to keep, in case we ever drop this skip we would need this for sure.